language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverUtil.java

```diff
@@ -54,80 +49,144 @@
     /**
      * Get the hover content.
      *
-     * @param context        Hover operation context
-     * @param cursorPosition Cursor position
+     * @param context Hover operation context
      * @return {@link Hover} Hover content
      */
-    public static Hover getHover(HoverContext context, Position cursorPosition) throws TokenOrSymbolNotFoundException {
+    public static Hover getHover(HoverContext context) {
         Optional<SemanticModel> semanticModel = context.workspace().semanticModel(context.filePath());
         if (semanticModel.isEmpty()) {
             return HoverUtil.getDefaultHoverObject();
         }
-        // TODO: Migrate to the latest
-        Token tokenAtCursor = TokensUtil.findTokenAtPosition(context, cursorPosition);
-        context.setTokenAtCursor(tokenAtCursor);
 
-        return HoverUtil.getDefaultHoverObject();
+        Position cursorPosition = context.getCursorPosition();
+        LinePosition linePosition = LinePosition.from(cursorPosition.getLine(), cursorPosition.getCharacter());
+        Optional<Symbol> symbolAtCursor = semanticModel.get()
+                .symbol(context.filePath().getFileName().getFileName().toString(), linePosition);
+        if (symbolAtCursor.isEmpty()) {
+            return HoverUtil.getDefaultHoverObject();
+        }
+
+        switch (symbolAtCursor.get().kind()) {
```

Comment:
Since the `Symbol.docAttachment();` is at `Symbol` level, can't we have a single function to get markup content? Easier and minimum changes required to maintain against future lang changes.

Reply:
We have to have separate functions since each of the construct behaves differently. For an instance, function's docs and the record's docs are completely different when considered to the class. Instead of having a single method with number of checks to bypass each case.